l10n_ro_hr/hr_employee.py

```diff
@@ -20,51 +20,151 @@
 #
 ##############################################################################
 
-from openerp import models, fields, api, _
 from datetime import datetime
+from openerp import models, fields, api, _
+from openerp.exceptions import ValidationError
 
+# luat de pe wikipedia: 
+# http://ro.wikipedia.org/wiki/Cod_numeric_personal#JJ
+pob = {
+    '01': u'Alba',
+    '02': u'Arad',
+    '03': u'Argeș',
+    '04': u'Bacău',
+    '05': u'Bihor',
+    '06': u'Bistrița-Năsăud',
+    '07': u'Botoșani',
+    '08': u'Brașov',
+    '09': u'Brăila',
+    '10': u'Buzău',
+    '11': u'Caraș-Severin',
+    '12': u'Cluj',
+    '13': u'Constanța',
+    '14': u'Covasna',
+    '15': u'Dâmbovița',
+    '16': u'Dolj',
+    '17': u'Galați',
+    '18': u'Gorj',
+    '19': u'Harghita',
+    '20': u'Hunedoara',
+    '21': u'Ialomița',
+    '22': u'Iași',
+    '23': u'Ilfov',
+    '24': u'Maramureș',
+    '25': u'Mehedinți',
+    '26': u'Mureș',
+    '27': u'Neamț',
+    '28': u'Olt',
+    '29': u'Prahova',
+    '30': u'Satu Mare',
+    '31': u'Sălaj',
+    '32': u'Sibiu',
+    '33': u'Suceava',
+    '34': u'Teleorman',
+    '35': u'Timiș',
+    '36': u'Tulcea',
+    '37': u'Vaslui',
+    '38': u'Vâlcea',
+    '39': u'Vrancea',
+    '40': u'București',
+    '41': u'București S.1',
+    '42': u'București S.2',
+    '43': u'București S.3',
+    '44': u'București S.4',
+    '45': u'București S.5',
+    '46': u'București S.6',
+    '51': u'Călărași',
+    '52': u'Giurgiu',
+}
 
-class hr_employee_care(models.Model):
-    _name = 'hr.employee.care'
-    _description = "Employee person in care"
-
-    employee_id = fields.Many2one('hr.employee', 'Employee', required=True)
-    name = fields.Char(
-        'Person care name', required=True, help='Person in care name')
-    ssnid = fields.Char('SSN No', required=True, help='Social Security Number')
-    relation = fields.Selection([('husband', 'Husband'),
-                                 ('wife', 'Wife'),
-                                 ('child', 'Child'),
-                                 ('firstdegree', 'First degree relationship'),
-                                 ('secdegree', 'Second degree relationship')],
-                                string='Person relation', required=True)
+def validate_cnp(cnp):
+    if len(cnp) != 13:
+        return False
+    cnpc = map(lambda x: int(x), cnp)
+    numc = [2, 7, 9, 1, 4, 6, 3, 5, 8, 2, 7, 9]
+    c = sum(map(lambda x: x[0]*x[1], zip(cnp[:-1], numc))) % 11
+    if c > 9: c = 1
+    if c != cnpc[-1]:
+        return False
+    return True
 
+class hr_employee_related(models.Model):
+    _name = 'hr.employee.related'
+    _description = "Employee person in care or are coinsured"
 
-class hr_employee_coinsured(models.Model):
-    _name = 'hr.employee.coinsured'
-    _description = "Employee co-insured persons"
+    @api.one
+    @api.onchange('ssnid')
+    @api.constrains('ssnid')
+    def _validate_ssnid(self):
+        if not validate_cnp(self.ssnid):
```

Comment:
Aici trebuie pus si daca e completat altfel cel putin la mine cand adaug o linie da eroare pentru ca e bool:

 if self.ssnid and not validate_cnp(self.ssnid):